### PR TITLE
feat: env override draft changes

### PR DESCRIPTION
**Pull request by sohel-rp**
# Description

Updates in env deployment template override view related config protection

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

All tests have been performed manually

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/EnvironmentOverride/DeploymentTemplateOverride.tsx [hunk 1]
```diff
@@ -80,40 +79,170 @@ export default function DeploymentTemplateOverride({
...
+                    },
+                }
+
+                dispatch({
+                    type: DeploymentConfigStateActionTypes.multipleOptions,
+                    payload,
+                })
+
+                if (state.selectedChart.name === ROLLOUT_DEPLOYMENT || state.selectedChart.name === DEPLOYMENT) {
+                    // updateTemplateFromBasicValue(envOverrideValues)
```
Comment (vivek-devtron) [inline src/components/EnvironmentOverride/DeploymentTemplateOverride.tsx:234]: Remove commented code if not required

**Review by vivek-devtron — APPROVED** [review]

File: src/components/EnvironmentOverride/DeploymentTemplateOverride.tsx [hunk 2]
```diff
@@ -687,91 +697,91 @@ function DeploymentTemplateOverrideForm({
...
-                    />
-                )}
+                <DeploymentConfigFormCTA
+                    loading={state.loading || state.chartConfigLoading}
+                    isEnvOverride={true}
+                    disableButton={!state.duplicate}
+                    disableCheckbox={!state.duplicate}
+                    showAppMetricsToggle={
+                        true
+                        // state.charts &&
```
Comment (vivek-devtron) [inline src/components/EnvironmentOverride/DeploymentTemplateOverride.tsx:762]: Remove commented code

**Review by vivek-devtron — APPROVED** [review]

File: src/components/EnvironmentOverride/DeploymentTemplateOverride.tsx (hunk 2 above)
Comment (sohel-rp) [reply src/components/EnvironmentOverride/DeploymentTemplateOverride.tsx]: Done

File: src/components/EnvironmentOverride/DeploymentTemplateOverride.tsx (hunk 1 above)
Comment (sohel-rp) [reply src/components/EnvironmentOverride/DeploymentTemplateOverride.tsx:236]: This will be covered in upcoming PR

**Pull request merged**